rs_workflows/common.py

```diff
@@ -0,0 +1,451 @@
+    logger.addHandler(console_handler)
+    return logger
+
+
+def check_status(endpoint, filename):
+    """Check the status of a file download from the specified rs-server endpoint.
+
+    This function sends a GET request to the rs-server endpoint with the filename as a query parameter
+    to retrieve the status of the file download. If the response is successful and contains a 'status'
+    key in the JSON response, the function returns the corresponding download status enum value. If the
+    response is not successful or does not contain the 'status' key, the function returns a FAILED status.
+
+    Args:
+        endpoint (str): The rs-server endpoint URL to query for the file status.
+        filename (str): The name of the file for which to check the status.
+
+    Returns:
+        EDownloadStatus: The download status enum value based on the response from the endpoint.
+
+    """
+    # TODO: check the status for a certain timeout if http returns NOK ?
+    try:
+        response = requests.get(
+            endpoint,
+            params={"name": filename},
+            timeout=ENDPOINT_TIMEOUT,
+        )
+
+        eval_response = response.json()
+        if (
+            response.ok
+            and "name" in eval_response.keys()
+            and filename == eval_response["name"]
+            and "status" in eval_response.keys()
+        ):
+            return EDownloadStatus(eval_response["status"])
+
+    except requests.exceptions.RequestException as e:
+        print(f"Request exception caught when calling for status endpoint: {e}")
+
+    return EDownloadStatus.FAILED
+
+
+def update_stac_catalog(url: str, user: str, mission: str, stac_file_info: dict):
+    # TODO ! Implement this when the catalog PUT endpoint will be ready
+    """
+    Each time a chunk is downloaded, publish it on catalog (with STAC
+    metadata returned in the first step + the file downloaded in S3 bucket)
+    RS-Server /catalog/rs-ops/collections/sx_chunk/items/{chunkid}
+    """
+    catalog_endpoint = url + f"/catalog/{user}/collections/{mission}/items/{stac_file_info}"
+    print(f"Endpoint to be used to insert the item info  within the catalog: {catalog_endpoint}")
+    # response = requests.put(catalog_endpoint, params=stac_file_info)
+    return True
+
+
+class PrefectCommonConfig:  # pylint: disable=too-few-public-methods
+    """Common configuration to Prefect tasks and flows.
+    Base class for configuration to prefect tasks and flows that ingest files from different stations (cadip, adgs...)
+
+    Attributes:
+        user (str): The user associated with the configuration.
+        url (str): The URL for the station.
+        station (str): The station identifier.
+        mission (str): The mission identifier.
+        tmp_download_path (str): The temporary download path.
+        s3_path (str): The S3 path for storing downloaded files.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
```

Comment:
Maybe here you could use a @dataclass instead, see: https://www.dataquest.io/blog/how-to-use-python-data-classes/

or as you did it for GetKeysFromS3Config

Reply:
I tried with @dataclass in the beginning, but it seems somehow the prefect flow doesn't like it. For the task, there wasn't any problem, but the flow complained with an error (don't remember what error).